### PR TITLE
Update ViT tutorial

### PR DESCRIPTION
**Pull request by vfdev-5**
First merge https://github.com/jax-ml/jax-ai-stack/pull/115 before merging this PR

Description
- [Updated ViT tutorial to fine-tune the model on Food101 dataset](https://github.com/jax-ml/jax-ai-stack/commit/478cf9395dd3f578f9997b0f103c9500d127ee70)

### CODE REVIEW

File: docs/JAX_Vision_transformer.md [hunk 1]
```diff
@@ -14,8 +14,7 @@ kernelspec:
 
 # Vision Transformer with JAX & FLAX
 
-
-In this tutorial we implement from scratch Vision Transformer (ViT) model based on the paper by Dosovitskiy et al: [An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale](https://arxiv.org/abs/2010.11929). We will train this model on [Food 101](https://huggingface.co/datasets/ethz/food101) dataset.
+In this tutorial we implement from scratch Vision Transformer (ViT) model based on the paper by Dosovitskiy et al: [An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale](https://arxiv.org/abs/2010.11929). We load the ImageNet pretrained weights and finetune this model on [Food 101](https://huggingface.co/datasets/ethz/food101) dataset.
```
Comment (melissawm) [inline docs/JAX_Vision_transformer.md:17]:
```suggestion
In this tutorial we implement from scratch a Vision Transformer (ViT) model based on the paper by Dosovitskiy et al: [An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale](https://arxiv.org/abs/2010.11929). We load the ImageNet pretrained weights and finetune this model on [Food 101](https://huggingface.co/datasets/ethz/food101) dataset.
```

File: docs/JAX_Vision_transformer.md [hunk 2]
```diff
@@ -162,9 +162,155 @@ class TransformerEncoder(nnx.Module):
...
+        assert dst_value.value.mean() == src_value.mean(), (dst_value.value, src_value.mean())
+
+    assert len(nonvisited) == 0, nonvisited
+    nnx.update(dst_model, nnx.State.from_flat_path(flax_model_params_fstate))
+
+
+vit_inplace_copy_weights(src_model=tf_model, dst_model=model)
+```
+
+Let's check the pretrained weights of our model and compare with the reference model results
```
Comment (melissawm) [inline docs/JAX_Vision_transformer.md:270]:
```suggestion
Let's check the pretrained weights of our model and compare with the reference model results:
```

File: docs/JAX_Vision_transformer.md [hunk 3]
```diff
@@ -162,9 +162,155 @@ class TransformerEncoder(nnx.Module):
...
+    f"Reference model:\n{tf_model.config.id2label[ref_class_idx]}\nP={nnx.softmax(logits, axis=-1)[0, ref_class_idx]:.3f}"
+)
+axs[0].imshow(image)
+axs[1].set_title(
+    f"Our model:\n{tf_model.config.id2label[pred_class_idx]}\nP={nnx.softmax(output, axis=-1)[0, pred_class_idx]:.3f}"
+)
+axs[1].imshow(image)
+```
+
+Now let's replace the classifier with a smaller fully-connected layer returning 20 classes instead of 1000
```
Comment (melissawm) [inline docs/JAX_Vision_transformer.md:308]:
```suggestion
Now, let's replace the classifier with a smaller fully-connected layer returning 20 classes instead of 1000:
```

File: docs/JAX_Vision_transformer.md [hunk 4]
```diff
@@ -567,12 +714,9 @@ for i in range(num_samples):
 
 ## Further reading
 
-In this tutorial we implemented from scratch Vision Transformer model and trained it on a subset of Food 101 dataset. Trained model shows 67% classification accuracy. Next steps could be to finetune hyperparameters like the learning rate and train for more epochs.
+In this tutorial we implemented from scratch Vision Transformer model and finetuned it on a subset of Food 101 dataset. Trained model shows almost perfect classification accuracy.
```
Comment (melissawm) [inline docs/JAX_Vision_transformer.md:717]:
```suggestion
In this tutorial we implemented from scratch a Vision Transformer model and finetuned it on a subset of Food 101 dataset. The trained model shows almost perfect classification accuracy.
```

**Review by melissawm — APPROVED** [review]
Tiny nits, otherwise looks good. Cheers!

**Review by jakevdp — APPROVED** [review]
Thanks!
